sympy/polys/polytools.py

```diff
@@ -4480,8 +4480,8 @@ def degree(f, gen=0):
          expression like func = %s, e.g. degree(func, gen = %s) instead of
          degree(func, gen = %s).
         ''' % (f, next(ordered(f.free_symbols)), gen)))
-
-    return Integer(p.degree(gen))
+    result = p.degree(gen)
+    return Integer(result) if result is int else S.NegativeInfinity
```

Comment:
`result is int` doesn't check if result is an integer. This should be
```suggestion
    return Integer(result) if isinstance(result, int) else S.NegativeInfinity
```

Although it would be simpler to just return a SymPy type, like `return sympify(result)`.

Reply:
Actually it would make more sense for Poly.degree to return an `Integer`. It already returns -oo, so I think it should always return a SymPy type. Then no conversion would be needed in this function.

Reply:
You are right. Corrected the isinstance. I have left the hardcoded conversion to reduce overhead.